plugins/rebalance/rest.go

```diff
@@ -0,0 +1,323 @@
+package rebalance
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+        "github.com/gluster/glusterd2/pkg/errors"
+	rebalanceapi "github.com/gluster/glusterd2/plugins/rebalance/api"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func createRebalanceInfo(volname string, req *rebalanceapi.StartReq) *rebalanceapi.RebalInfo {
+	return &rebalanceapi.RebalInfo{
+		Volname:     volname,
+		RebalanceID: uuid.NewRandom(),
+                Status:      rebalanceapi.Started,
+		Cmd:         checkCmd(req),
+		CommitHash:  setCommitHash(),
+		RebalStats:       rebalanceapi.RebalNodeStatus{},
+	}
+}
+
+func rebalanceStartHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	var req rebalanceapi.StartReq
+
+//  TODO : How do I pass these options?
```

Comment:
You can pass these to the transaction by setting them in the transaction context.

Reply:
`curl -i -XPOST <rebalance-url> -d '{"force": true}'`

- `-i` to display response headers
- "force" should be defined as `json:"force"` in StartReq